Apps/Contoso.WinUI.Desktop.Demo/Contoso.WinUI.Desktop.Demo/App.xaml.cs

```diff
@@ -0,0 +1,153 @@
+
+            // Set userId.
+            var userId = localSettings.Values[Constants.KeyUserId] as string;
+            if (!string.IsNullOrEmpty(userId))
+            {
+                AppCenter.SetUserId(userId);
+            }
+            Crashes.HasCrashedInLastSessionAsync().ContinueWith(hasCrashed =>
+            {
+                Log("Crashes.HasCrashedInLastSession=" + hasCrashed.Result);
+            });
+            Crashes.GetLastSessionCrashReportAsync().ContinueWith(task =>
+            {
+                Log("Crashes.LastSessionCrashReport.StackTrace=" + task.Result?.StackTrace);
+            });
+        }
+
+        private static void Log(string message)
+        {
+            var timestamp = DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss.fff", CultureInfo.InvariantCulture);
+            System.Diagnostics.Debug.WriteLine($"{timestamp} [AppCenterPuppet] Info: {message}");
+        }
+
+        private static IEnumerable<ErrorAttachmentLog> GetErrorAttachmentsHandler(ErrorReport report)
+        {
+            return GetErrorAttachments();
+        }
+
+        public static IEnumerable<ErrorAttachmentLog> GetErrorAttachments()
+        {
+            List<ErrorAttachmentLog> attachments = new List<ErrorAttachmentLog>();
+
+            // Text attachment
+            var localSettings = ApplicationData.Current.LocalSettings;
+            var textAttachments = localSettings.Values[Constants.KeyTextErrorAttachments] as string;
+            if (!string.IsNullOrEmpty(textAttachments))
+            {
+                attachments.Add(
+                    ErrorAttachmentLog.AttachmentWithText(textAttachments, "text.txt"));
+            }
+
+            // Binary attachment
+            var fileAttachments = localSettings.Values[Constants.KeyFileErrorAttachments] as string;
+            if (!string.IsNullOrEmpty(fileAttachments))
+            {
+                if (File.Exists(fileAttachments))
+                {
+                    var fileName = new FileInfo(fileAttachments).Name;
+                    var provider = new FileExtensionContentTypeProvider();
+                    if (!provider.TryGetContentType(fileName, out var contentType))
+                    {
+                        contentType = "application/octet-stream";
+                    }
+                    var fileContent = File.ReadAllBytes(fileAttachments);
+                    attachments.Add(ErrorAttachmentLog.AttachmentWithBinary(fileContent, fileName, contentType));
+                }
+                else
+                {
+                    localSettings.Values[Constants.KeyFileErrorAttachments] = null;
+                }
+            }
+            return attachments;
+        }
+
+        /// <summary>
+        /// Invoked when the application is launched normally by the end user.  Other entry points
+        /// will be used such as when the application is launched to open a specific file.
```

Comment:
```suggestion
        /// will be used when the application is launched by OS, for example, to open a specific file.
```

Reply:
Btw, don't we need to implement other launch types as well?

Reply:
this comment was copied from other our test apps, lets keep it as it is